pkg/apis/core/v1beta1/types_common.go

```diff
@@ -20,12 +20,12 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 type ErrorCode string
 
 const (
-	// ErrorInfraUnauthorized indicates that the last error occurred due to invalid infrastructure credentials.
+	// ErrorInfraUnauthenticated indicates that the last error occurred due to the client request not being completed because it lacks valid authentication credentials for the requested resource.
 	// It is classified as a non-retryable error code.
-	ErrorInfraUnauthorized ErrorCode = "ERR_INFRA_UNAUTHORIZED"
-	// ErrorInfraInsufficientPrivileges indicates that the last error occurred due to insufficient infrastructure privileges.
+	ErrorInfraUnauthenticated ErrorCode = "ERR_INFRA_UNAUTHENTICATED"
+	// ErrorInfraUnauthorized indicates that the last error occurred due to the server understanding the request but refusing to authorize it.
 	// It is classified as a non-retryable error code.
-	ErrorInfraInsufficientPrivileges ErrorCode = "ERR_INFRA_INSUFFICIENT_PRIVILEGES"
```

Comment:
Should we for backwards-compatibility purposes keep the `ERR_INFRA_INSUFFICIENT_PRIVILEGES` code while mapping it to the same as `ERR_INFRA_UNAUTHORIZED` but deprecating it? This will allow us to merge this PR without "breaking" existing usage of these codes (e.g. in the Gardener Dashboard). After some releases we can then finally drop this deprecated code from the code base. WDYT?

Reply:
Yes, makes sense.